apps/explorer/src/pages/transaction-result/SendReceiveView.tsx

```diff
@@ -1,56 +1,239 @@
+                                    Recipients
+                                </Heading>
+                            </div>
+                            {recipient.map((add: string, idx: number) => (
+                                <div key={idx}>
+                                    <>
+                                        <div className={styles.oneaddress}>
+                                            <div className={styles.doneicon}>
+                                                <DoneIcon />
+                                            </div>
+                                            <Longtext
+                                                text={add}
+                                                category="addresses"
+                                                isLink={true}
+                                                alttext={add}
+                                            />
+                                        </div>
+                                        {amount?.[idx] && (
+                                            <Amount
+                                                amount={amount![idx]}
+                                                label={
+                                                    isSuccess && coinList
+                                                        ? coinList[idx]
+                                                        : ''
+                                                }
+                                            />
+                                        )}
+                                    </>
+                                </div>
+                            ))}
+                        </div>
                     )}
-                </h3>
-            </div>
-            <div
-                className={cl([
-                    styles.txaddresssender,
-                    data.recipient?.length ? styles.recipient : '',
-                ])}
-            >
-                <Longtext
-                    text={data.sender}
-                    category="addresses"
-                    isLink={true}
-                />
-                {data.recipient && (
-                    <ul className={styles.txrecipents}>
-                        {data.recipient.map((add: string, idx: number) => (
-                            <li key={idx}>
-                                <Longtext
-                                    text={add}
-                                    category="addresses"
-                                    isLink={true}
-                                    alttext={add}
-                                />
-                            </li>
-                        ))}
-                    </ul>
-                )}
+                </div>
             </div>
+        </>
+    );
+}
+
+function Amount({ amount, label }: { amount: bigint; label: string }) {
+    const coinBoilerPlateRemoved = /^0x2::coin::Coin<(.+)>$/.exec(label)?.[1];
```

Comment:
I think that `Coin.getCoinTypeArg` from the SDK does this for you.

Reply:
I've exported `COIN_TYPE_ARG_REGEX` from the SDK and used this here. As you will see in the SDK, the function `Coin.getCoinTypeArg` expects the `ObjectData` not the `string` type.

Reply:
Sounds fine for now, although we should update this.